test/Microsoft.AspNetCore.Mvc.Razor.Test/ViewComponentTagHelperDescriptorProviderTest.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.Razor.Host;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor
+{
+    public class ViewComponentTagHelperDescriptorProviderTest
+    {
+        [Fact]
+        public void GetDescriptorsFromAssembly_ReturnsAllViewComponentsAsTagHelperDescriptors()
+        {
+            // Act
+            var descriptors = ViewComponentTagHelperDescriptorProvider.GetDescriptorsFromAssembly("Microsoft.AspNetCore.Mvc.Razor.Test");
+
+            // Assert
+            Assert.Collection(descriptors,
```

Comment:
This is kinda brittle. It relies on never adding another view component type to this assembly. Maybe do a `Contains` check instead?